### PR TITLE
[jsx-sort-default-props] Bug fix for one case with spread

### LINKED ISSUE #2178: Add option to ignore spread operator in sort-default-props
With `foo = { bar: baz }`.
```
{ ...foo, bar } = { bar }
```
And
```
{ bar, ...for } = { baz }
```

So I think, an option to ignore spread operator in sort default props will be great.
Correct example
```
{ a: 'a', c: 'c', ...foo, b: 'b', d: 'd' }
```
Incorrect example
```
{ a: 'a', ...foo, c: 'c', b: 'b', d: 'd' }
``` 

Plus, it allow to develop a fixer (asked [here](https://github.com/yannickcr/eslint-plugin-react/issues/2062)) without breaking the code.

### PR DESCRIPTION
**Pull request by VincentLanglet**
Closes #2178

I thought an option was needed, but instead it was a bug

### CODE REVIEW

File: tests/lib/rules/jsx-sort-default-props.js [hunk 1]
```diff
@@ -314,12 +314,17 @@ ruleTester.run('jsx-sort-default-props', rule, {
...
+      '  b: "b"',
+      '};',
       'const First = (props) => <div />;',
       'export const propTypes = {',
       '    a: PropTypes.any,',
+      '    b: PropTypes.string,',
       '    z: PropTypes.string,',
       '};',
       'export const defaultProps = {',
+      '    ...defaults,',
```
Comment (ljharb) [inline tests/lib/rules/jsx-sort-default-props.js]: instead of modifying this test, could you add a new one?

Comment (VincentLanglet) [reply tests/lib/rules/jsx-sort-default-props.js]: Done

File: lib/rules/jsx-sort-default-props.js [hunk 2]
```diff
@@ -96,7 +96,7 @@ module.exports = {
      */
     function checkSorted(declarations) {
       declarations.reduce((prev, curr, idx, decls) => {
-        if (/SpreadProperty$/.test(curr.type)) {
+        if (/SpreadProperty$/.test(curr.type) || /SpreadElement$/.test(curr.type)) {
```
Comment (ljharb) [inline lib/rules/jsx-sort-default-props.js]:
```suggestion
        if (/Spread(?:Property|Element)$/.test(curr.type)) {
```

**Comment by VincentLanglet** [comment]
@ljharb Done :)

**Pull request merged**